rfhub2/api/endpoints/collections.py

```diff
@@ -24,3 +26,29 @@ def get_collection(*, repository: CollectionRepository = Depends(get_collection_
                    id: int):
     collection: Optional[Collection] = repository.get(id)
     return or_404(collection)
+
+
+@router.post("/", response_model=Collection, status_code=201)
+def create_collection(*, repository: CollectionRepository = Depends(get_collection_repository),
+                      collection: CollectionUpdate):
+    db_collection = DBCollection(**collection.dict())
+    return repository.add(db_collection)
+
+
+@router.put("/{id}/", response_model=Collection)
+def update_collection(*, repository: CollectionRepository = Depends(get_collection_repository),
+                      id: int,
+                      collection_update: CollectionUpdate):
+    db_collection: DBCollection = or_404(repository.get(id))
+    updated: DBCollection = repository.update(db_collection, collection_update.dict(skip_defaults=True))
+    return updated
+
+
+@router.delete("/{id}/")
+def delete_collection(*, repository: CollectionRepository = Depends(get_collection_repository),
+                      id: int):
+    deleted: int = repository.delete(id)
```

Comment:
I can see in code that this method is deleting collection https://github.com/pbylicki/rfhub-new/blob/master/rfhub2/db/repository/collection_repository.py#L18 using ```return self.session.query(Collection).options(selectinload(Collection.keywords))```. Does this do cascade delete so that keywords are dropped first, and then collection itself?

Reply:
Ok, from SQLAlchemy docs I see it does cascade delete. Splenind!

Reply:
Yes, it does, but it is worth remembering that in order to make it work on SQLite, you need this:
https://github.com/pbylicki/rfhub-new/blob/master/rfhub2/db/session.py#L18